src/mac.cpp

```diff
@@ -154,6 +154,57 @@ CFArrayRef CreateChildrenArray(AXUIElementRef element) {
   return children;
 }
 
+CFStringRef GetNestedText(AXUIElementRef element, CFMutableArrayRef textChildren) {
```

Comment:
it's a little confusing how `GetNestedText` and `GetNestedTextFromChildren` are currently structured, since they're both recursive and iterate through children. I'm not sure I understand this code 100%, but I think there are 3 things going on:

1. get the text from leaf nodes
2. if a node isn't a leaf, build a flattened / 1D array of elements
3. join the flattened array into a single string

is there a way we can structure the code to make each function a little more narrow and clear?

Reply:
I changed the names of these functions to clarify what's happening. `GetNestedText` is now `GetLineText` and `GetNestedTextFromChildren` is now `GetLines`. `GetLineText` does pretty much what you laid out, recursing through the tree until a leaf node and stringing all of those leaves together with no space in between. `GetLines` runs `GetLineText` on each of the starting node's children and combines the results with newlines in between. If `GetLines` encounters a list element, it recurses once since the list element is just a container for the lines.

As an example, if I have the following message:

Here is a list of items:
* item 1
* item 2

the resulting tree looks like
```
* (text entry element)
    * (group)
        * Here is a list of items: (text)
    * (list)
        * (group)
            * \u0006 (text)
            * item 1 (text)
        * (group)
            * \u0006 (text)
            * item 2 (text)
```

So `GetLines` will call `GetLineText` on each of the group elements (the ones within the list element will be reached by the recursive call in `GetLines`) and combine with newlines to get the correct state.